demux/utils/novaseq_samplesheet.py

```diff
@@ -0,0 +1,181 @@
+""" Create a samplesheet for Novaseq flowcells """
+
+import csv
+
+from .runparameters import NovaseqRunParameters
+from .samplesheet import Samplesheet
+
+
+class CreateNovaseqSamplesheet:
+    """ Create a raw sample sheet for Novaseq flowcells """
+
+    LIMS_KEYS = [
+        "fcid",
+        "lane",
+        "sample_id",
+        "sample_ref",
+        "index",
+        "index2",
+        "sample_name",
+        "control",
+        "recipe",
+        "operator",
+        "project",
+    ]
+
+    def __init__(
+        self, flowcell, indexlength, pad, raw_samplesheet, dummy_indexes, runs_dir
+    ):
+        self.flowcell = flowcell
+        self.indexlength = indexlength
+        self.pad = pad
+        self.raw_samplesheet = raw_samplesheet
+        self.dummy_indexes = dummy_indexes
+        self.runparameters = NovaseqRunParameters(self.flowcell, runs_dir)
+
+    @property
+    def header(self):
+        """ Create the sample sheet header """
+        return list(Samplesheet.header_map.values())
+
+    @staticmethod
+    def get_project(project):
+        """ Only keeps the first part of the project name"""
+        return project.split(" ")[0]
+
+    @staticmethod
+    def reverse_complement(dna):
+        """ Generates the reverse complement """
+        complement = {"A": "T", "C": "G", "G": "C", "T": "A"}
+        return "".join([complement[base] for base in dna[::-1]])
+
+    def add_dummy_indexes(self):
```

Comment:
None of these def use any args except self. Would it not be more readable and encapulated if all things used by the def is passed to the def. Self hides a lot and makes it hard to understand the "players" taking part in the function

Reply:
I don't think I understand the problem? The 'player' is the class instance, it modifies the class instance and returns it. I sue methods that only take `self` all the time.

Another reason I chose to do it this was is that it allows method chaining, which I thought was nice. See https://code.activestate.com/recipes/580616-python-method-chaining-examples/ again .for examples.

Reply:
I am just worrying that if you want start supplying args to these functions the oneline chaining will become much harder to read and understand. As it is I have no idea what parts of self will be used and modified since there are no args to the defs. I have to read the entire funtions to understand what will be done. Therefore, this is like a black box on self and how it is modified.